Applications/WebPortal/WebApp/library/source/pages/account_directory_page/local_account_directory_model.ts

```diff
@@ -0,0 +1,44 @@
+import * as Beam from 'beam';
+import { AccountDirectoryModel } from './account_directory_model';
+import { AccountEntry } from './account_entry';
+
+/** Implements an EntitlementsModel in memory. */
+export class LocalAccountDirectoryModel extends AccountDirectoryModel {
+
+  /** Constructs an new model.
+   * @param groupList - The account to represent.
+   * @param accountLists - The set of entitlements granted to the account.
+   */
+  constructor( groupList: Beam.Set<Beam.DirectoryEntry>,
+      accountLists: Beam.Map<Beam.DirectoryEntry, AccountEntry[]>) {
+    super();
+    this._isLoaded = false;
+    this._groupList = groupList;
```

Comment:
What if the person who passed the ```groupList``` to you decides to add or delete groups or modify the parameter they gave to you? That will introduce inconsistencies and bugs.